src/java/org/apache/cassandra/db/lifecycle/View.java

```diff
@@ -69,6 +69,7 @@
     public final List<Memtable> flushingMemtables;
     final Set<SSTableReader> compacting;
     final Set<SSTableReader> sstables;
+    final Map<String, SSTableReader> sstablesByFilename;
```

Comment:
This change worries me.

I don't think a change of this kind belongs in Cassandra proper: it's functionality very specific to CNDB that nothing else needs or uses. Is it possible to move it to the CNDB code? Possibly in the form of an expiring weak-reference cache.

Reply:
Yes, that's a good point.

From what I can tell looking at the usages in cndb, it is always accessed through `Tracker.getLiveSSTable(String)`, where, if iiuc, the `View` instance may change over time thus making it impractical to try and cache on the cndb side.

Alternatively, if it's not a significant performance hit or critical path, cndb can perform the lookup by name, something like:
```
    static SSTableReader getDataTrackerLiveSSTable(Tracker tracker, String sstableName)
    {
        return tracker.getLiveSSTables().stream()
                .filter(ssTableReader -> ssTableReader.getFilename().equals(sstableName))
                .findFirst()
                .orElse(null);
    }
```

In any case, perhaps it is best to remove this part and open a new ticket to implement something on the cndb side.